tests/Collection.test.ts

```diff
@@ -44,33 +45,94 @@ describe("Collection", () => {
-    });
+    const datasets = [
+      { name: "dataset1", path: "dir1" },
+      { name: "dataset2", path: "dir2/nested/some_dir/" },
+      { name: "dataset3", path: "dir3/data.json" },
+      { name: "dataset4", path: "https://b.com/dataset4/" },
+      { name: "dataset5", path: "https://b.com/dataset5/custom.json" },
+    ];
+
+    const collectionFormats: { version: string; collectionJson: object }[] = [
+      { version: "v0.0.0", collectionJson: datasets },
+      {
+        version: "v1.1.0",
+        collectionJson: {
+          datasets: datasets,
+          metadata: {
+            name: "c_name",
+            description: "c_description",
+            author: "c_author",
+            collectionVersion: "c_collectionVersion",
+            lastModified: "2024-01-01T00:00:00Z",
+            dateCreated: "2023-01-01T00:00:00Z",
+            revision: 15,
+            writerVersion: "v1.1.0",
+          },
+        },
+      },
+    ];
+
+    for (const { version, collectionJson } of collectionFormats) {
+      describe(version, () => {
+        it("collects datasets", async () => {
+          const url = "https://e.com/collection.json";
+          const mockFetch = makeMockFetchMethod(url, collectionJson);
+          const collection = await Collection.loadCollection(url, mockFetch);
+
+          expect(collection.getDatasetKeys().length).to.equal(5);
+          expect(collection.getDatasetKeys()).to.deep.equal([
+            "dataset1",
+            "dataset2",
+            "dataset3",
+            "dataset4",
+            "dataset5",
+          ]);
+        });
+
+        it("should parse dataset paths correctly", async () => {
+          const url = "https://e.com/collection.json/";
+          const mockFetch = makeMockFetchMethod("https://e.com/collection.json", collectionJson);
+          const collection = await Collection.loadCollection(url, mockFetch);
+
+          expect(collection.getAbsoluteDatasetPath("dataset1")).to.equal("https://e.com/dir1/manifest.json");
+          expect(collection.getAbsoluteDatasetPath("dataset2")).to.equal(
+            "https://e.com/dir2/nested/some_dir/manifest.json"
+          );
+          expect(collection.getAbsoluteDatasetPath("dataset3")).to.equal("https://e.com/dir3/data.json");
+          expect(collection.getAbsoluteDatasetPath("dataset4")).to.equal("https://b.com/dataset4/manifest.json");
+          expect(collection.getAbsoluteDatasetPath("dataset5")).to.equal("https://b.com/dataset5/custom.json");
+        });
+
+        it("retrieves metadata", async () => {
+          const url = "https://e.com/collection.json";
+          const mockFetch = makeMockFetchMethod(url, collectionJson);
+          const collection = await Collection.loadCollection(url, mockFetch);
+
+          if (semver.lte(version, "v0.0.0")) {
+            expect(collection.metadata).deep.equals({});
+            return;
```

Comment:
this is a personal preference, but I would have made these two different tests: one that sends in v0.0.0 data and checks for the empty object. and one that sends in v1.1.0 data and has everything below. and the description would be "it returns an empty object if the version is 0.0.0" since it's not really retrieving any metadata in this case. Also is there a possibility of there not being a version or do you add it somewhere (I'm starting here in the review)

Reply:
The versions are defined in a big array up above, so here it's checking versions of the data that I've explicitly labeled as `v0.0.0`, `v1.0.0`, and `v1.1.0`. While I think it makes sense to split these into different tests as the tests are now, I'm also anticipating that more backwards-compatible versions will get added in the future. Structuring this test as it is now (where it progressively tests more and more fields) makes sense to me for that specific use case.

Reply:
Mm, taking another look at it, I think it makes sense to split these up for now. (or at least, to make `v0.0.0` its own case!)